### ISSUE TITLE
Decide if drag&drop room lists should come back

### ISSUE THREAD

**Issue by turt2live** [element-hq/element-web#8614]
Currently we don't support dragging rooms between tags on the redesign. We should either support it or rip out all the code which supports it.

**Comment by Cadair** [comment]
+1 to bringing it back

**Comment by shirishag75** [comment]
+1 to that as well.

**Comment by turt2live** [comment]
(before we end up with a billion +1 comments, please use github reactions on the top comment instead)

**Comment by jimmycuadra** [comment]
Is there another way to reorder rooms within the same section? I used to use drag and drop to do this. It's not clear from the description if there's another way to do this or whether it's currently impossible in the redesign.

**Comment by ara4n** [comment]
currently impossible; workaround is to use https://tags.t2bot.io

**Comment by lampholder** [comment]
Discussing this with Nad and Amandine yesterday - we're all keen for drag and drop to come back _if we can guarantee that we will execute and maintain it well_.

If we do it, I want us to do it so that:
- it has finely tuned heuristics to avoid accidental drag and drop
- it defends against the state changing underneath you mid drag
- it defends against the state changing just before you drag
- it works well in all browsers
- it does not introduce any performance regressions
- it supports undo to avoid making mistakes that are hard to correct
- it has a sophisticated local echo awareness so that any changes made via drag and drop don't glitch back and forth
- it can feedback to the user if whatever changes you made _don't_ end up being persisted remotely for whatever reason

My personal bias is that we implement our UX such that it works stably and well _without_ drag and drop, and then once that solid baseline is in place, consider then whether it's worth the additional investment to add a robust dnd layer on top.

**Comment by alphapapa** [comment]
This issue seems relevant: https://github.com/vector-im/riot-web/issues/8723  It claims that a "labs" setting exists to re-enable room user-tags, but I can't find such a labs setting.

Please bring back this feature.  I used it extensively to group related rooms.  Now they're all in one big "rooms" group, which is useless to me.

**Comment by jryans** [comment]
> This issue seems relevant: #8723 It claims that a "labs" setting exists to re-enable room user-tags, but I can't find such a labs setting.

(As mentioned in #8723, it's available on https://riot.im/develop only at the moment.)

**Comment by JesseNaranjo** [comment]
> @lampholder https://github.com/vector-im/riot-web/issues/8614#issuecomment-472849425
> If we do it, I want us to do it so that:
> ...
>     * it supports undo to avoid making mistakes that are hard to correct
> ...

I agree with everything on the list, except the "undo" line. I think undo in a drag-n-drop situation should be fairly easy to do manually - especially if you can only drag one item *at a time*.

I am not opposed to the undo feature, but if implementing "drag-n-drop undo" requirement is a big hurdle, I would rather see everything else implemented without this.

**Comment by alphapapa** [comment]
> * it has finely tuned heuristics to avoid accidental drag and drop
> * it defends against the state changing underneath you mid drag
> * it defends against the state changing just before you drag
> * it works well in all browsers
> * it does not introduce any performance regressions
> * it supports undo to avoid making mistakes that are hard to correct
> * it has a sophisticated local echo awareness so that any changes made via drag and drop don't glitch back and forth
> * it can feedback to the user if whatever changes you made _don't_ end up being persisted remotely for whatever reason

I don't think any features in any software I've used fulfill that list of requirements.  For example, file managers don't have finely tuned heuristics to avoid accidental drag and drop, and that's in a context where such accidents could cause data loss.

We're talking about room tags here.  The worst that could happen is that a user would have to drag and drop a room into a tag again.

**Comment by swedneck** [comment]
Perhaps one option could be to make the order lockable from the settings? I at least don't want to re-order my rooms very often, and would be fine with having to untick an option to be able to re-order them and then re-tick the option.

**Comment by BloodyIron** [comment]
I'm in favour of click and dragging between sections, such as 1:1 or rooms to favourites, etc. Like, that's just so convenient, why _not_ have it? IMO

**Comment by turt2live** [comment]
I've closed/linked a bunch of issues above in favour of this one. If/when we do bring the feature back, the above issues should be considered.

**Comment by BloodyIron** [comment]
Drag & Drop stuff is always welcome!

**Comment by Flam3z** [comment]
+1 dragging and dropping makes organizing so much more easier.

**Comment by ahayes** [comment]
> (before we end up with a billion +1 comments, please use github reactions on the top comment instead)

That could work if there was a single clear question to react to. I would like to give a +1 to supporting drag and drop and a -1 to ripping that support out.

I would also like to be able to drag a room between spaces because I'm sure that somehow in the past I have had success in getting People (DM rooms?) from home into spaces, but by default they always seem to get created in home, even if I used a space's People + button to start a new chat.

**Comment by paboum** [comment]
Definitely it would help to sort rooms inside a space in other ways than alphabetically and drag-n-dropping them between spaces.

**Comment by forrestwilkins** [comment]
Has there been any progress made for this issue since 2022? I'd love to see this feature brought back.

**Comment by t3chguy** [comment]
Dragging between sections would likely be desirable, dragging to specify a manual sort order is unlikely desirable with Product

**Comment by daniellekirkwood** [comment]
We support drag and drop of rooms into sections, and drag and drop of sections to a new order. Closing this ticket
